src/_arraykit.c

```diff
@@ -383,7 +383,28 @@ dtype_from_element(PyObject *Py_UNUSED(m), PyObject *arg)
 
     // Integers
     if (PyLong_CheckExact(arg)) {
-        return (PyObject*)PyArray_DescrFromType(NPY_LONG);
+        int overflow;
+        npy_int64 v = PyLong_AsLongLongAndOverflow(arg, &overflow);
+        if (v != -1 && overflow == 0) {
+            // Prioritize fastest case first
+            return (PyObject*)PyArray_DescrFromType(NPY_LONG); // (-2**63, 2**63)
```

Comment:
```suggestion
            return (PyObject*)PyArray_DescrFromType(NPY_LONG);  // [-2**63, 2**63)
```

Reply:
The choice of brackets versus parentheses was deliberate. `[` and `]` are inclusive, while `(` and `)` are exclusive.